.github/workflows/cmake.yml

```diff
@@ -92,3 +81,86 @@ jobs:
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: 11
+      - name: Set up Python for gcovr
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install gcovr
+        run: |
+          pip install gcovr==5.0
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Download and set up sonar-scanner
+        env:
+          SONAR_SCANNER_DOWNLOAD_URL: https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${{ env.SONAR_SCANNER_VERSION }}-linux.zip
+        run: |
+          mkdir -p $HOME/.sonar
+          curl -sSLo $HOME/.sonar/sonar-scanner.zip ${{ env.SONAR_SCANNER_DOWNLOAD_URL }}
+          unzip -o $HOME/.sonar/sonar-scanner.zip -d $HOME/.sonar/
+          echo "$HOME/.sonar/sonar-scanner-${{ env.SONAR_SCANNER_VERSION }}-linux/bin" >> $GITHUB_PATH
+      - name: Download and set up build-wrapper
+        env:
+          BUILD_WRAPPER_DOWNLOAD_URL: ${{ env.SONAR_SERVER_URL }}/static/cpp/build-wrapper-linux-x86.zip
+        run: |
+          curl -sSLo $HOME/.sonar/build-wrapper-linux-x86.zip ${{ env.BUILD_WRAPPER_DOWNLOAD_URL }}
+          unzip -o $HOME/.sonar/build-wrapper-linux-x86.zip -d $HOME/.sonar/
+          echo "$HOME/.sonar/build-wrapper-linux-x86" >> $GITHUB_PATH
+      - name: Run CMake
+        run: |
+          cmake \
+            -Dlibsettings_ENABLE_PYTHON=0 \
+            -DCMAKE_POSITION_INDEPENDENT_CODE=1 \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -DCMAKE_C_FLAGS="--coverage" \
+            -DCMAKE_CXX_FLAGS="--coverage"
+      - name: Build with coverage
+        run: |
+          build-wrapper-linux-x86-64 --out-dir ${{ env.BUILD_WRAPPER_OUT_DIR }} make VERBOSE=1
+      - name: Generate coverage report
+        run: |
+          gcovr -v --sonarqube -o coverage.xml
+      - name: Cache SonarCloud analysis files
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.ANALYSIS_CACHE_DIR }}
+          key: ${{ runner.os }}-analysis
+          restore-keys: ${{ runner.os }}-analysis
+      - name: Run sonar-scanner
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
+          sonar-scanner --define sonar.host.url="${{ env.SONAR_SERVER_URL }}" \
+          --define sonar.organization="${{ env.SONAR_ORG }}" \
+          --define sonar.projectKey="${{ env.SONAR_PROJECT_KEY }}" \
+          --define sonar.sources="src,python" \
+          --define sonar.tests="tests" \
+          --define sonar.python.version="2.7,3.7,3.8,3.9,3.10,3.11" \
```

Comment:
Do we need all these Python versions?

Reply:
It matched what we are running the CI on. You can leave the version unset but then SonarCloud provides a warning that it won't do complete analysis. I don't actually know what it will change depending on the Python versions enabled.

Reply:
Ok, surprised to see 2.7 still on there

Reply:
I thought about removing 2.7 from CI but wasn't sure if we had kept it around for a reason, I can remove it we don't care about it.

Reply:
Probably just kept around because it isn't failing the build yet, I think it's safer to preemptively drop it though.